### PR TITLE
Add dark theme to the buttons

### LINKED ISSUE #3779: Base button is white on grey background
**Describe the bug**

In 2.31.0 the background of base button was changed from transparent to white. It makes the button visible on lightgrey backgrounds (like MAAS) when it was not visible before.

Screenshot 2021-06-08 at 08 51 20

**Expected behavior**

Screenshot 2021-06-08 at 08 55 01

**Context**

This blocks MAAS update to 2.31.0: https://github.com/canonical-web-and-design/maas-ui/pull/2736

### PR DESCRIPTION
**Pull request by bartaz**
## Done

WIP

Demo: https://vanilla-framework-3807.demos.haus/docs/examples/patterns/buttons/dark

Fixes #3779

## QA

- Open [demo](https://vanilla-framework-3807.demos.haus/docs/examples/patterns/buttons/dark)
- Check if dark buttons render as expected
- Review updated documentation:
  - https://vanilla-framework-3807.demos.haus/docs/patterns/buttons#theming
  - https://vanilla-framework-3807.demos.haus/docs/component-status

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

- [x] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [x] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical-web-and-design/vanilla-framework/releases/latest), following semver convention:
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical-web-and-design/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [x] Any changes to component class names (new patterns, variants, removed or added features) should be listed on the [component status page](https://github.com/canonical-web-and-design/vanilla-framework/blob/master/templates/docs/component-status.md).
- [x] Documentation side navigation should be updated with the relevant labels.

## Screenshots

Screenshot 2021-06-30 at 17 16 12

### CODE REVIEW

File: scss/_patterns_buttons.scss [hunk 1]
```diff
@@ -92,16 +117,48 @@ $hover-background-opacity-percentage: $hover-background-opacity-amount * 100%;
     @include vf-focus($color-focus-positive);
   }
 
+  %p-button--positive-dark {
+    $color-positive-dark: adjust-color($color-positive, $lightness: -10%);
```
Comment (bartaz) [inline scss/_patterns_buttons.scss:121]: @lyubomir-popov This is just provisional darker version of positive color, I'll let you adjust it properly.

File: scss/_patterns_buttons.scss [hunk 2]
```diff
@@ -117,17 +174,63 @@ $hover-background-opacity-percentage: $hover-background-opacity-amount * 100%;
     @include vf-focus($color-focus-negative);
   }
 
+  %p-button--negative-dark {
+    $color-negative-dark: adjust-color($color-negative, $lightness: -10%);
```
Comment (bartaz) [inline scss/_patterns_buttons.scss:178]: @lyubomir-popov This is just provisional darker version of negative color, I'll let you adjust it properly.

File: scss/_patterns_buttons.scss [hunk 3]
```diff
@@ -117,17 +174,63 @@ $hover-background-opacity-percentage: $hover-background-opacity-amount * 100%;
...
+    }
+
     @extend %vf-button-white-success-icon;
   }
 }
 
 @mixin vf-button-base {
-  %p-button--base {
-    @extend %vf-button-base;
+  %p-button--base-light {
```
Comment (bartaz) [inline scss/_patterns_buttons.scss:218]: @lyubomir-popov Now with dark/light themes we should be able to change background on base button back to transparent?

File: templates/docs/examples/patterns/buttons/dark.html [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    <p>
+        <button class="p-button--positive is-dark">Positive button</button>
+        <button class="p-button--positive is-dark" disabled>Positive button disabled</button>
+    </p>
+    <p>
+        <button class="p-button--negative is-dark">Negative button</button>
+        <button class="p-button--negative is-dark" disabled>Negative button disabled</button>
+    </p>
+    <p>
+        <button class="p-button--brand is-dark">Brand button</button>
```
Comment (bartaz) [inline templates/docs/examples/patterns/buttons/dark.html:25]: @lyubomir-popov I haven't added dark theme to brand button yet - how would we create it from unknown color?

Comment (lyubomir-popov) [reply templates/docs/examples/patterns/buttons/dark.html:25]: We should have two variables, you can set both to the same current value

File: scss/_patterns_buttons.scss (hunk 2 above)
Comment (lyubomir-popov) [reply scss/_patterns_buttons.scss:178]: add !default; maybe?

**Comment by bartaz** [comment]
@lyubomir-popov Can you review and update colors where needed?

**Comment by lyubomir-popov** [comment]
@spencerbygraves @bartaz what do you think of these?
The darker ones are more traditional, similar to what github and apple do. The lighter ones stick out more, and borrow from the material design theory that colors on dark should be desaturated and lightend to become more readable.

Full benchmaring doc [here](https://www.figma.com/file/geUaGt9GJhGZs5E8NCiaGv/Dark-theme-primary%2Fsecondary-button-proposal?node-id=0%3A1)

**Comment by bartaz** [comment]
@lyubomir-popov Personally for me the top (option A) feels more like a dark variant of our buttons. With desaturated bottom ones I wasn't sure if they are disabled.

**Comment by spencerbygraves** [comment]
@lyubomir-popov I also prefer the top ones but I think they could be lighter.

**Comment by lyubomir-popov** [comment]
@bartaz just spoke to @spencerbygraves, can we use #008013 and #A11223 for positive/negative colours in the default state? Then apply the hover/active state percentages to those.

image

**Review by sowasred2012 — APPROVED** [review]
Looks great!

**Comment by lyubomir-popov** [comment]
Design +1 and percy approved @bartaz

**Pull request merged**